### ISSUE TITLE
JSON.pretty_generate generates different output to MRI for at least an empty hash

### ISSUE THREAD

**Issue by chrisseaton** [jruby/jruby#6338]
```ruby
require 'json'
puts JSON.pretty_generate({})
```

```
% chruby 2.5.7
% ruby -v test.rb 
ruby 2.5.7p206 (2019-10-01 revision 67816) [x86_64-darwin19]
{
}
% chruby jruby 9.2.11.1
% ruby -v test.rb
jruby 9.2.11.1 (2.5.7) 2020-03-25 b1f55b1a40 OpenJDK 64-Bit Server VM 25.252-b14 on 1.8.0_252-b14 +jit [darwin-x86_64]
{

}
```

**Comment by headius** [comment]
I see in the other issue that it may be a behavior of json-pure, but in JRuby it ought to be loading the extension. Could be a behavioral change that was done to the C ext and never to the pure-Ruby or Java exts.

**Comment by headius** [comment]
Could you re-file this with the json repo? We do not maintain that source here.

**Comment by chrisseaton** [comment]
Do you know how can I tell what version of JSON has _really_ been loaded? I don't see this error in the Java source, so not sure how I'm loading the pure version, if that's what is happening.

**Comment by chrisseaton** [comment]
Sorry cancel - I was confused and the bug was in Java as well and I've fixed it in https://github.com/flori/json/pull/449.

**Comment by headius** [comment]
@chrisseaton Thanks for the patch! I can't review it or do anything to move it forward unfortunately, but once it's released we can update JRuby.

**Comment by headius** [comment]
Actually, I can review... will do so now. Someone else will have to merge and release, though.

**Comment by chrisseaton** [comment]
That's merged now. I'll close on anticipation you'll pick up the new version soon enough.

**Comment by headius** [comment]
This has not been addressed yet since the change in flori/json#449 has not been released. We do not ship prerelease sources for default gems.

@chrisseaton In the future please don't close issues in our tracker unless they are actually resolved and targeted against a specific release. This one flew under the radar for 9.2.14 because it was closed without a release.

**Comment by headius** [comment]
Marked for 9.3 since this is trivial and the fix has not yet been released. Could target 9.2.15 if the fix is released and we do that version before a 9.3 release.

**Comment by headius** [comment]
Updated json for both 9.2 and 9.3 so this should be resolved now.
